diff --git a/build.rs b/build.rs
--- a/build.rs
+++ b/build.rs
@@ -1,4 +1,7 @@
 fn main() {
+    println!("cargo:rerun-if-env-changed=RUSTDESK_APP_NAME");
+    println!("cargo:rerun-if-env-changed=RUSTDESK_FORCE_WEBSOCKET");
+
     let out_dir = format!("{}/protos", std::env::var("OUT_DIR").unwrap());
 
     std::fs::create_dir_all(&out_dir).unwrap();
diff --git a/src/config.rs b/src/config.rs
index 6fe1004..11fba8a 100644
--- a/src/config.rs
+++ b/src/config.rs
@@ -69,7 +69,11 @@ lazy_static::lazy_static! {
     static ref ONLINE: Mutex<HashMap<String, i64>> = Default::default();
     pub static ref PROD_RENDEZVOUS_SERVER: RwLock<String> = RwLock::new("".to_owned());
     pub static ref EXE_RENDEZVOUS_SERVER: RwLock<String> = Default::default();
-    pub static ref APP_NAME: RwLock<String> = RwLock::new("RustDesk".to_owned());
+    pub static ref APP_NAME: RwLock<String> = RwLock::new(
+        option_env!("RUSTDESK_APP_NAME")
+            .unwrap_or("RustDesk")
+            .to_owned()
+    );
     static ref KEY_PAIR: Mutex<Option<KeyPair>> = Default::default();
     static ref USER_DEFAULT_CONFIG: RwLock<(UserDefaultConfig, Instant)> = RwLock::new((UserDefaultConfig::load(), Instant::now()));
     pub static ref NEW_STORED_PEER_CONFIG: Mutex<HashSet<String>> = Default::default();
diff --git a/src/websocket.rs b/src/websocket.rs
--- a/src/websocket.rs
+++ b/src/websocket.rs
@@ -423,7 +423,9 @@
     };
     let protocol = if is_domain {
         let api_server = Config::get_option("api-server");
-        if api_server.starts_with("https") {
+        if option_env!("RUSTDESK_FORCE_WEBSOCKET") == Some("Y")
+            || api_server.starts_with("https")
+        {
             "wss"
         } else {
             "ws"
@@ -480,9 +482,10 @@
         assert_eq!(check_ws("127.0.0.1:21115"), "ws://127.0.0.1:21118");
         assert_eq!(check_ws("127.0.0.1:21116"), "ws://127.0.0.1:21118");
         assert_eq!(check_ws("127.0.0.1:21117"), "ws://127.0.0.1:21119");
-        assert_eq!(check_ws("rustdesk.com:21115"), "ws://rustdesk.com/ws/id");
-        assert_eq!(check_ws("rustdesk.com:21116"), "ws://rustdesk.com/ws/id");
-        assert_eq!(check_ws("rustdesk.com:21117"), "ws://rustdesk.com/ws/relay");
+        let protocol = if option_env!("RUSTDESK_FORCE_WEBSOCKET") == Some("Y") { "wss" } else { "ws" };
+        assert_eq!(check_ws("rustdesk.com:21115"), format!("{protocol}://rustdesk.com/ws/id"));
+        assert_eq!(check_ws("rustdesk.com:21116"), format!("{protocol}://rustdesk.com/ws/id"));
+        assert_eq!(check_ws("rustdesk.com:21117"), format!("{protocol}://rustdesk.com/ws/relay"));
         // set relay-server without port
         Config::set_option("relay-server".to_string(), "127.0.0.1".to_string());
         Config::set_option(
