diff --git a/build.rs b/build.rs
index 43e3282..e472bea 100644
--- a/build.rs
+++ b/build.rs
@@ -1,4 +1,6 @@
 fn main() {
+    println!("cargo:rerun-if-env-changed=RUSTDESK_APP_NAME");
+
     let out_dir = format!("{}/protos", std::env::var("OUT_DIR").unwrap());
 
     std::fs::create_dir_all(&out_dir).unwrap();
diff --git a/src/config.rs b/src/config.rs
index 6fe1004..11fba8a 100644
--- a/src/config.rs
+++ b/src/config.rs
@@ -69,7 +69,11 @@ lazy_static::lazy_static! {
     static ref ONLINE: Mutex<HashMap<String, i64>> = Default::default();
     pub static ref PROD_RENDEZVOUS_SERVER: RwLock<String> = RwLock::new("".to_owned());
     pub static ref EXE_RENDEZVOUS_SERVER: RwLock<String> = Default::default();
-    pub static ref APP_NAME: RwLock<String> = RwLock::new("RustDesk".to_owned());
+    pub static ref APP_NAME: RwLock<String> = RwLock::new(
+        option_env!("RUSTDESK_APP_NAME")
+            .unwrap_or("RustDesk")
+            .to_owned()
+    );
     static ref KEY_PAIR: Mutex<Option<KeyPair>> = Default::default();
     static ref USER_DEFAULT_CONFIG: RwLock<(UserDefaultConfig, Instant)> = RwLock::new((UserDefaultConfig::load(), Instant::now()));
     pub static ref NEW_STORED_PEER_CONFIG: Mutex<HashSet<String>> = Default::default();
